src/App.tsx

```diff
@@ -1,39 +1,63 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './Components/Clock';
 
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+export type State = {
+  hasClock: boolean
+  clockName: string,
+};
 
-        <span className="Clock__time">
-          {date.toLocaleTimeString()}
-        </span>
+class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
+
+  newName = 0;
+
+  componentDidMount() {
+    this.newName = window.setInterval(this.nameChanger, 3300);
+    document.addEventListener('click', (event: MouseEvent) => {
+      if (event.button === 0) {
+        this.setState({ hasClock: true });
+        this.newName = window.setInterval(this.nameChanger, 3300);
+      }
+    });
+
+    document.addEventListener('contextmenu', event => {
+      event.preventDefault();
+      this.setState({ hasClock: false });
+      // eslint-disable-next-line no-console
+      console.log(this.state.clockName);
+      clearInterval(this.newName);
+    });
+  }
+
+  nameChanger = () => {
+    this.setState({ clockName: getRandomName() });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1 className="App__title">React clock</h1>
+        <div className="App__info">
+          <div>
+            <strong>{this.state.clockName}</strong>
```

Comment:
use destructuring, you wrote `this.state.clockName` 3 times, while you could just use `clockName`